biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java

```diff
@@ -366,8 +367,18 @@ private Processor loadRepositories() throws Exception {
 			// Map the repository names...
 
 			Map<String,Repository> repoNameMap = new HashMap<String,Repository>(allRepos.size());
-			for (Repository repo : allRepos)
-				repoNameMap.put(repo.toString(), repo);
+			for (Repository repo : allRepos) {
+				String name = repo.toString();
```

Comment:
The toString call can be defered to a final else block to avoid calling if not needed. And also to avoid double calling for wrapper plugin.